contracts/main/stablecoin-core/StableSwapModule.sol

```diff
@@ -178,22 +164,30 @@ contract StableSwapModule is PausableUpgradeable, ReentrancyGuardUpgradeable, IS
         emit LogStableSwapPauseState(false);
     }
 
-    function emergencyWithdraw(address _account) external override nonReentrant onlyOwnerOrGov{
-        require(paused(),"emergencyWithdraw/not-paused");
-        tokenBalance[address(token)] = 0;
-        tokenBalance[address(stablecoin)] = 0;
+    function emergencyWithdraw(address _account) external override nonReentrant onlyOwnerOrGov whenPaused {
+        require(_account != address(0), "withdrawFees/empty-account");
+        tokenBalance[token] = 0;
+        tokenBalance[stablecoin] = 0;
         token.safeTransfer(_account, token.balanceOf(address(this)));
-        address(stablecoin).safeTransfer(_account, address(stablecoin).balanceOf(address(this)));
+        stablecoin.safeTransfer(_account, stablecoin.balanceOf(address(this)));
         emit LogEmergencyWithdraw(_account);
     }
 
     function _udpateAndCheckDailyLimit(uint256 _amount) internal {
-        if(block.timestamp - lastUpdate >= ONE_DAY){
+        if (block.timestamp - lastUpdate >= ONE_DAY) {
             lastUpdate = block.timestamp;
-            remainingDailySwapAmount= dailySwapLimit;
+            remainingDailySwapAmount = dailySwapLimit;
         }
-        require(remainingDailySwapAmount >= _amount,"_udpateAndCheckDailyLimit/daily-limit-exceeded");
+        require(remainingDailySwapAmount >= _amount, "_udpateAndCheckDailyLimit/daily-limit-exceeded");
         remainingDailySwapAmount -= _amount;
         emit LogRemainingDailySwapAmount(remainingDailySwapAmount);
     }
+
+    function _convertDecimals(
```

Comment:
this function was added for amount normalization, on order to have all calculations with 18 decimals